examples/step-27/doc/results.dox

```diff
@@ -209,7 +209,71 @@ include estimating the smoothness not on single cells, but cell assemblies or
 sub-optimal refinement criterion.
+
+
+
+<a name="extensions"></a>
+<h3>Possibilities for extensions</h3>
+
+<h4>Different hp-decision strategies</h4>
+
+This tutorial demonstrates only one particular strategy to decide between h- and
+p-adaptation. In fact, there are many more ways to automatically decide on the
+adaptation type from which a couple are already implemented in deal.II:
+<ul>
+  <li><i>Fourier coefficient decay:</i> This is the strategy currently
+  implemented in this tutorial. For more information on this strategy, see
+  the general documentation of the SmoothnessEstimator::Fourier namespace.</li>
+
+  <li><i>Legendre coefficient decay:</i> This strategy is quite similar
+  to the current one, but uses Legendre series expansion rather than the
+  Fourier one: instead of sinusoids as basis functions, this strategy uses
+  Legendre polynomials. The transition from the Fourier strategy to
+  the Legendre one is quite simple: You just need to change the series
+  expansion class and the corresponding smoothness estimation function
+  to be part of the proper namespaces FESeries::Legendre and
+  SmoothnessEstimator::Legendre. For the theoretical background of this
+  strategy, consult the general documentation of the
+  SmoothnessEstimator::Legendre namespace, as well as @cite mavriplis1994hp ,
+  @cite eibner2007hp and @cite davydov2017hp.</li>
+
+  <li><i>Refinement history:</i> The last strategy is quite different
+  from the other two. In theory, we know how the error will converge
+  after changing the discretization of the function space. With
+  h-refinement, the solution converges algebraically as already pointed
+  out in step-7. It the solution is sufficiently smooth, though, we
+  expect that the solution will converge exponentially with increasing
+  polynomial degree of the finite element. We can compare a proper
+  prediction of the error with the actual error in the following step to
+  see verify if our choice of adaptation type was justified. The transition
+  to this strategy is a bit more complicated. For this, we need an
+  initialization step with pure h- or p-refinement and we need to transfer
+  the predicted errors over adapted meshes. The extensive documentation
+  of the hp::Refinement::predict_error() function describes not only the
+  theoretical details of this approach, but also presents a blueprint on
+  how to implement this strategy in your code. Note that this strategy
+  is only suited for time-independent problems or to create an initial
+  mesh for time-dependent problems. For more information, see
+  @cite melenk2001hp .</li>
+</ul>
+
+Try implementing one of these strategies into this tutorial and observe the
+subtle changes to the results. You will notice that all strategies are
+capable of identifying the singularities near the reentrant corners and
+will perform h-refinement in these regions, while preferring p-refinement
+in the bulk domain.
```

Comment:
I think it would be fair to cite your PhD thesis somewhere here. You *did* implement all of these things, and it makes sense to point the reader to a place where they can find the kind of comparison you're suggesting here.

Reply:
Yes, we could do this or point to the parallel hp-paper (if we decide to describe them there, too).

Reply:
For now, just reference your thesis. We can always add other references at a later time.